### PR TITLE
Make tryAliasThisCast() a nested function and remove redundant checks

### PR DESCRIPTION
**Pull request by BorisCarvajal**
Because all its parameters live in the parent.
And the checks are the same as the following lines after where I'm inserting the function, [here](https://github.com/BorisCarvajal/dmd/blob/5ea8bcc67771367b0c115624461a75d28770006f/src/dmd/dcast.d#L1563-L1567).

### CODE REVIEW

File: src/dmd/dcast.d [hunk 1]
```diff
@@ -1467,27 +1467,6 @@ Type toStaticArrayType(SliceExp e)
...
-private Expression tryAliasThisCast(Expression e, Scope* sc, Type tob, Type t1b, Type t)
-{
-    Expression result;
-    AggregateDeclaration t1ad = isAggregate(t1b);
-    if (!t1ad)
-        return null;
-
-    AggregateDeclaration toad = isAggregate(tob);
-    if (t1ad == toad || !t1ad.aliasthis)
-        return null;
```
Comment (WalterBright) [inline src/dmd/dcast.d:1480]: What happened to lines 1473-1480?

Comment (BorisCarvajal) [reply src/dmd/dcast.d:1480]: Look at the URL in the first comment, they were redundant and had 0 coverage according to test suite.

Comment (RazvanN7) [reply src/dmd/dcast.d:1480]: Maybe that could be done in the PR that is fixing the issue.

**Comment by RazvanN7** [comment]
@BorisCarvajal To move forward with this, just move the function without touching it and you can delete the uncovered lines in https://github.com/dlang/dmd/pull/12168 . It makes more sense that way from a git history perspective.

**Review by RazvanN7 — APPROVED** [review]

**Pull request merged**